contrib/codegen/templates/sets/sets.gotmpl

```diff
@@ -103,4 +104,14 @@ func (s {{ $kindLowerCamel }}Set) Intersection(set {{ $resource.Kind }}Set) {{ $
     return New{{ $resource.Kind }}Set({{ $kindLowerCamel }}List...)
 }
 
+
+func (s {{ $kindLowerCamel }}Set) Find(id ezkube.ResourceId) (*{{ $import_prefix }}.{{ $resource.Kind }}, error) {
```

Comment:
`id ezkube.ResourceId` doesn't feel like the right arg for this function. Most times this function will most likely be called with the user only having the string information. My recommendation here would be to either, have a simple companion struct which implements `ezkube.ResourceId`
for instance:
```
type ResourceKey struct {
    Name string
    Namespace string
    Cluster string
}
// implemented methods
```
This way find can be called similar to how other Gets/Reads work throughout sk and skv2

Reply:
you can use a `ClusterObjectRef` as the argument for Find, or an existing object. I prefer the interface because of the flexibility it offers